arm-dns/2016-04-01/swagger/dns.json

```diff
@@ -0,0 +1,1202 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/RecordSet"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "RecordSets"
+        ],
+        "operationId": "RecordSets_CreateOrUpdate",
+        "description": "Creates a RecordSet within a DNS zone.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "zoneName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the zone without a terminating dot."
+          },
+          {
+            "name": "recordType",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The type of DNS record.",
+            "enum": [
+              "A",
+              "AAAA",
+              "CNAME",
+              "MX",
+              "NS",
+              "PTR",
+              "SOA",
+              "SRV",
+              "TXT"
+            ],
+            "x-ms-enum": "RecordType"
+          },
+          {
+            "name": "relativeRecordSetName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the RecordSet, relative to the name of the zone.",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RecordSetCreateOrUpdateParameters"
+            },
+            "description": "Parameters supplied to the CreateOrUpdate operation."
+          },
+          {
+            "name": "IfMatch",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "description": "The etag of RecordSet."
+          },
+          {
+            "name": "IfNoneMatch",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "description": "Defines the If-None-Match condition. Set to '*' to force Create-If-Not-Exist. Other values will be ignored."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/RecordSet"
+            }
+          },
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/RecordSet"
+            }
+          }
+        }
```

Comment:
Is this a long running operation? If so then please apply the `"x-ms-long-running": true` extension.